GitUI/UserControls/RevisionGrid.cs

```diff
@@ -427,6 +428,78 @@ private void RevisionsKeyDown(object sender, KeyEventArgs e)
             }
         }
 
+        private void RevisionsKeyUp(object sender, KeyEventArgs e)
+        {
+            if (Revisions.RowCount < LastRowIndex || LastRowIndex < 0 || Revisions.RowCount == 0)
```

Comment:
I would introduce a function for this condition. While it is pretty easy to know what it does (but only if you know the true meaning of the `LastRowIndex`), it is better to name it explicitly. Reading a code I don't want to guess what the programmer wanted to do. If it is said explicitly by declaring a dedicated method or variable then I can validate if the condition is well formed (the implementation is consistent with the declaration) otherwise I have to figure out what the condition is supposed to do before checking if it is correct.